File: frontend/src/types/weather.ts
export interface WeatherPoint {
  lat: number;
  lon: number;
  forecast_time: string;
  temperature_c?: number | null;
  relative_humidity_percent?: number | null;
  wind_speed_ms: number;
  wind_gust_ms: number;
  precipitation_mm: number;
  cloud_cover_percent?: number | null;
  weather_code?: number | null;
}

File: app/schemas/weather.py
from datetime import datetime

from pydantic import BaseModel


class WeatherPointResponse(BaseModel):
    lat: float
    lon: float
    forecast_time: datetime
    temperature_c: float | None = None
    relative_humidity_percent: float | None = None
    wind_speed_ms: float
    wind_gust_ms: float
    precipitation_mm: float
    cloud_cover_percent: float | None = None
    weather_code: int | None = None
